packages/@pollyjs/adapter-node-http/src/utils/transport-wrapper.js

```diff
@@ -0,0 +1,212 @@
+
+    const { headers, statusCode } = res;
+
+    return {
+      headers,
+      statusCode,
+      data
+    };
+  }
+
+  async respond(req, response) {
+    Object.keys(req[LISTENERS]).forEach(eventName => {
+      const listeners = req[LISTENERS][eventName];
+
+      listeners.forEach(listener => req.on(eventName, listener));
+    });
+
+    const fakeSocket = { readable: false };
+    const msg = new http.IncomingMessage(fakeSocket);
+
+    msg.statusCode = response.statusCode;
+    req.emit('response', msg);
+
+    for (const h in response.headers) {
+      msg.headers[h] = response.headers[h];
+    }
+
+    let body = response.body;
+
+    if (typeof body === 'undefined') {
+      body = [];
+    }
+
+    msg.emit('data', Buffer.from(body));
+    msg.emit('end');
+
+    req.emit('prefinish');
+    req.emit('finish');
+  }
+
+  createRequestWrapper() {
+    const wrapper = this;
+    const { adapter, transport } = wrapper;
+    const nativeRequest = nativeRequestMapping.get(transport);
+
+    return (...args) => {
+      /**
+       * args could be (url, options, callback) or (options, callback) depending on nodejs version
+       * We don't need to wrap callback we can just emit 'response' event on ClientRequest
+       * and it's more reliable to collect request information from ClientRequest instance
+       * We don't need to store args
+       */
+
+      const req = nativeRequest.call(transport, ...args);
+
+      const chunks = [];
+
+      const originalWrite = req.write;
+
+      req.write = (chunk, encoding, callback) => {
+        // TODO : handle encoding
+        chunks.push(chunk);
+
+        originalWrite.call(req, chunk, encoding, callback);
+      };
+
+      // This can be called multiple times with the same request instance
+      // so make sure we only save the original `end` method.
+      req[NATIVE_END] = req[NATIVE_END] || req.end;
+      req.end = (chunk, encoding, callback) => {
+        if (typeof chunk === 'function') {
+          callback = chunk;
+          chunk = null;
+        } else if (typeof encoding === 'function') {
+          callback = encoding;
+          encoding = null;
+        }
+
+        if (chunk) {
+          req.write(chunk, encoding);
+        }
+
+        if (typeof callback === 'function') {
+          // we don't call original `end` yet but no need to carry callback around when we do
+          // this is what happens in original `end`
+          req.once('finish', callback);
+        }
+
+        req[LISTENERS] = ['response'].reduce((acc, eventName) => {
+          const listeners = req.listeners(eventName);
+
+          // unsubscribe these listeners, so that we break the connection between caller and `req`
+          // until we decide what to do in next steps
+          req.removeAllListeners(eventName);
+
+          acc[eventName] = listeners;
+
+          return acc;
+        }, {});
+
+        const headers =
+          typeof req.getHeaders === 'function'
```

Comment:
`req.getHeaders()` is not a valid method in Node 6. I'm not sure if the changes I've made are correct since the only valid property I could access was `req._headers`. 

@yasinuslu if you could confirm this logic, that would be appreciated 😄

Reply:
@offirgolan I couldn't find where it gets assigned in `nodejs` source code but i think it will be fine :smile: